mkfat32.zig

```diff
@@ -0,0 +1,361 @@
+            16 * GB...32 * GB - 1 => std.math.max(16384, bytes_per_sector) / bytes_per_sector,
+            32 * GB...2 * TB - 1 => std.math.max(32768, bytes_per_sector) / bytes_per_sector,
+            else => Error.TooLarge,
+        });
+    }
+
+    ///
+    /// Get the default volume name of a FAT image: "NO NAME    ".
+    ///
+    /// Return: [11]u8
+    ///     The default volume name.
+    ///
+    pub fn getDefaultVolumeName() [11]u8 {
+        return "NO NAME    ".*;
+    }
+
+    ///
+    /// Get the default image size: 33548800 (~32KB). This is the recommended minimum image size
+    /// for FAT32. 0xFFF5 (valid cluster values) * 512 (getDefaultSectorSize).
+    ///
+    /// Return: u32
+    ///     The smallest recommended FAT32 image size.
+    ///
+    pub fn getDefaultImageSize() u32 {
+        return 0xFFF5 * @as(u32, getDefaultSectorSize());
+    }
+
+    ///
+    /// Create the FAT32 serial number. This is generated from the date. Reference:
+    /// https://www.digital-detective.net/documents/Volume%20Serial%20Numbers.pdf
+    ///
+    /// Arguments:
+    ///     IN current_time: SerialTime - The current date and time.
+    ///
+    /// Return: u32
+    ///     Serial number for a new FAT32 image.
+    ///
+    pub fn createSerialNumber(current_time: SerialTime) u32 {
+        const p1 = (@as(u16, current_time.month) << 8) | current_time.day;
+        const p2 = (@as(u16, current_time.second) << 8) | current_time.millisecond_10;
+        const p3 = (@as(u16, current_time.hour) << 8) | current_time.minute;
+
+        const p4 = p1 + p2;
+        const p5 = p3 + current_time.year;
+
+        return (@as(u32, p4) << 16) + p5;
+    }
+
+    ///
+    /// Write the FSInfo and backup FSInfo sector to a image file. This uses a valid FAT32 boot
+    /// sector header for creating the FSInfo sector.
+    ///
+    /// Argument:
+    ///     IN/OUT image: File        - The image file to write the FSInfo to.
+    ///     IN fat32_header: Header   - A valid FAT32 boot header for creating the FSInfo sector.
+    ///     IN free_cluster_num: u32  - The number of free data clusters on the image.
+    ///     IN next_free_cluster: u32 - The next free data cluster to start looking for when writing
+    ///                                 files.
+    ///
+    /// Error File.WriteError || File.SeekError
+    ///     File.WriteError - If there is an error when writing to the image. See File.WriteError
+    ///     File.SeekError  - If there is an error when seeking the image. See File.SeekError
+    ///
+    pub fn writeFSInfo(image: File, fat32_header: Header, free_cluster_num: u32, next_free_cluster: u32) (File.WriteError || File.SeekError)!void {
+        // Seek to the correct location
+        try image.seekableStream().seekTo(fat32_header.fsinfo_sector * fat32_header.bytes_per_sector);
+
+        // First signature
+        try image.writer().writeIntLittle(u32, 0x41615252);
```

Comment:
Can `image.writer()` be stored in a variable instead of being called every time?

Reply:
In the std lib, when multiple writes are made, `writer()` is called each time. But it didn't make a difference when storing the write in a variable. ??

Reply:
I just think it would be a little cleaner.

Reply:
ok